frontend/src/__tests__/unit/testUtils/hooks.spec.ts

```diff
@@ -0,0 +1,187 @@
+import { renderHook as renderHookRTL } from '@testing-library/react';
+import { expectHook, renderHook, testHook } from './hooks';
+
+const useSayHello = (who: string, showCount = false) => {
+  const countRef = React.useRef(0);
+  countRef.current++;
+  return `Hello ${who}!${showCount && countRef.current > 1 ? ` x${countRef.current}` : ''}`;
+};
+
+const useSayHelloDelayed = (who: string, delay = 0) => {
+  const [speech, setSpeech] = React.useState('');
+  React.useEffect(() => {
+    const handle = setTimeout(() => setSpeech(`Hello ${who}!`), delay);
+    return () => clearTimeout(handle);
+  }, [who, delay]);
+  return speech;
+};
+
+describe('hook test utils', () => {
+  describe('comparison of hook tests', () => {
+    const useSayHello = (who: string) => `Hello ${who}!`;
+
+    it('use hook test utils', () => {
+      const renderResult = testHook(useSayHello, 'world');
+      expectHook(renderResult).toBe('Hello world!').toHaveUpdateCount(1);
+      renderResult.rerender('world');
+      expectHook(renderResult).toBe('Hello world!').toBeStable().toHaveUpdateCount(2);
+    });
+
+    it('use RTL only', () => {
+      let updateCount = 0;
+
+      const renderResult = renderHookRTL(
+        ({ who }: { who: string }) => {
+          updateCount++;
+          return useSayHello(who);
+        },
+        { initialProps: { who: 'world' } },
+      );
+      expect(renderResult.result.current).toBe('Hello world!');
+      expect(updateCount).toBe(1);
+
+      const prevValue = renderResult.result.current;
+
+      renderResult.rerender({
+        who: 'world',
+      });
+      expect(renderResult.result.current).toBe('Hello world!');
+      expect(renderResult.result.current).toBe(prevValue);
+      expect(updateCount).toBe(2);
+    });
+  });
+
+  it('use testHook for rendering', () => {
+    const renderResult = testHook(useSayHello, 'world');
+    expectHook(renderResult)
+      .toHaveUpdateCount(1)
+      .toBe('Hello world!')
+      .toEqual('Hello world!')
+      .toStrictEqual('Hello world!');
+
+    renderResult.rerender('world', false);
+
+    expectHook(renderResult)
+      .toHaveUpdateCount(2)
+      .toBe('Hello world!')
+      .toEqual('Hello world!')
+      .toStrictEqual('Hello world!')
+      .toBeStable();
+
+    renderResult.rerender('world', true);
+
+    expectHook(renderResult)
+      .toHaveUpdateCount(3)
+      .toBe('Hello world! x3')
+      .toEqual('Hello world! x3')
+      .toStrictEqual('Hello world! x3')
+      .toBeStable(false);
+  });
+
+  it('use renderHook for rendering', () => {
+    type Props = {
+      who: string;
+      showCount?: boolean;
+    };
+    const renderResult = renderHook(({ who, showCount }: Props) => useSayHello(who, showCount), {
+      initialProps: {
+        who: 'world',
+      },
+    });
+
+    expectHook(renderResult)
+      .toHaveUpdateCount(1)
+      .toBe('Hello world!')
+      .toEqual('Hello world!')
+      .toStrictEqual('Hello world!');
```

Comment:
Is this my test naivety? Why not go straight to strict equals? soft equals has value in any test where you also do a strict equals?

Reply:
This is simply testing _all_ the available functions on the api.

That beings said, we talked about it and think that `toStrictEqual` should always be used and therefore I will be removing `toEqual` from the API.